src/rgw/rgw_tracer.h

```diff
@@ -0,0 +1,55 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab ft=cpp
+
+#ifndef RGW_TRACER_H
+#define RGW_TRACER_H
+
+#define SIGNED_RIGHT_SHIFT_IS 1
+#define ARITHMETIC_RIGHT_SHIFT 1
+#include <jaegertracing/Tracer.h>
+#include "common/config.h"
+#include "common/ceph_context.h"
+
+namespace tracing {
+
+class TracerConfiguration : public md_config_obs_t {
+private:
+  bool enabled;
+
+public:
+  void init(CephContext* cct);
+  bool is_enabled() const;
+  const char** get_tracked_conf_keys() const override;
+  void handle_conf_change(const ConfigProxy& conf, const std::set<std::string>& changed) override;
+};
+
+class Tracer {
+private:
+  const static std::shared_ptr<opentracing::Tracer> noop_tracer;
+  std::shared_ptr<opentracing::Tracer> open_tracer;
+  static TracerConfiguration tracer_conf;
```

Comment:
why is the conf static?

Reply:
please see https://github.com/ceph/ceph/pull/42434#discussion_r677373205